src/LanguageServer.php

```diff
@@ -100,4 +120,42 @@ public function exit()
     {
         exit(0);
     }
+
+    /**
+     * Parses workspace files, one at a time.
+     *
+     * @param string $rootPath The rootPath of the workspace. Is null if no folder is open.
+     * @return void
+     */
+    private function indexProject(string $rootPath)
+    {
+        $dir = new \RecursiveDirectoryIterator($rootPath);
+        $ite = new \RecursiveIteratorIterator($dir);
+        $files = new \RegexIterator($ite, '/^.+\.php$/i', \RegexIterator::GET_MATCH);
+        $fileList = array();
+        foreach($files as $file) {
+            $fileList = array_merge($fileList, $file);
+        }
+        
+        $processFile = function() use (&$fileList, &$processFile, &$rootPath){
+            if ($file = array_pop($fileList)) {
+                
+                $uri = 'file://'.(substr($file, -1) == '/' || substr($file, -1) == '\\' ? '' : '/').str_replace('\\', '/', $file);
```

Comment:
We should make a helper function for this, will need it more often. I am for adding a `utils.php` with plain functions that are autoloaded through composer.json `files` property